### ISSUE TITLE
Panels check does not check sub directories

### ISSUE THREAD

**Issue by cbv08** [siemens/CtrlppCheck#70]
**Describe the bug**
When using the "check panels (static)" a report is made, this report only checks the panels directories for files. It does not check through the sub directories in panels. 

This also leads to the report falsely stating that the count of sub directories is zero

**To Reproduce**
1. Using version 1.0.1 stable release
2. have panels directory with sub directories with panels within
3. Click on 'check panels (static)'
4. open report see result missing panels from sub dir

**Expected behavior**
All panels within the "Panels dir" should be checked and the report should cover all of them

**Screenshots**

**Software version and environment (please complete the following information):**
 - OS: Windows 10
 - WinCC OA Version: 3.19
 - Version: 1.0.1

**Comment by jumoog** [comment]
CtrlppCheck is intended for Ctrl scripts only. You can extract your logic and paste it into Ctrl libraries.

**Comment by cbv08** [comment]
I mean that's fair. But there is a button called "Panels Check (Static)" in the main window. I tried tweaking it a bit in version 1.0.2 it does actually check through the panels in the sub directories. And gives some fairly interesting information. 

However as you rightly point out it is incabable of checking the logic contained within the panel. It just gives errors stating that it can't. However i don't think this invalidates my open issue. 

If the intention is that CtrlppCheck should never be used with panels, then i might suggest removing the previously described button.

Kind regards
Olav

**Comment by jumoog** [comment]
No, that doesn't invalidate your issue. If the button is there, it should work. I just wanted to give you a tip. Have a nice weekend!

**Comment by cbv08** [comment]
Ah okay, great, thank you for the tip, it is greatly appreciated! 

I hope it is okay if I ask you another question, it has been a bit hard to find information on the usage of Ctrlppcheck. Do you use the additions in Gedi or do you only use the command line version? Have a nice weekend!

Kind regards 
Olav

**Comment by dhoegerlETM** [comment]
Completely checking the CTRL code within UI panels is currently a known missing feature and already handled with #64.

Nevertheless, the "check panels" button executes a few other checks and must work irrespectively.

@mPokornyETM it seems we introduced some regression here - I use most recent 1.0.2 and get following log entries when double-clicking the report after "check panels".

WCCOAui      (1), 2023.11.08 22:07:04.287, CTRL, WARNING,    61, Cannot open file, 
    Module: Vision_3
    Panel: \WinCCOA_QualityChecks\panels\vision\QualityChecks\QG_Result.pnl [Panels (static)]
    Script: Initialize
    Line: 8, fileToString, <proj>/data/QualityGates/QgStaticCheck_Panels/1699477548/Result, The system cannot find the file specified.
WCCOAui      (1), 2023.11.08 22:07:04.342, CTRL, WARNING,    28/ctrl, int, Given datatype does not match with needed type 'mapping', 
    Module: Vision_3
    Panel: \WinCCOA_QualityChecks\panels\vision\QualityChecks\QG_Result.pnl [Panels (static)]
    Script: Initialize
    Line: 31, mappingHasKey
WCCOAui      (1), 2023.11.08 22:07:04.351, CTRL, SEVERE,    176, Uncaught exception in thread started at main: 
WCCOAui      (1), 2023.11.08 22:07:04.342, CTRL, SEVERE,     79, Index out of range, 
    Module: Vision_3
    Panel: \WinCCOA_QualityChecks\panels\vision\QualityChecks\QG_Result.pnl [Panels (static)]
    Script: Initialize
    Line: 31, map[root], root
	Stacktrace:
	     1: void main() at [Module: Vision_3 Panel: \WinCCOA_QualityChecks\panels\vision\QualityChecks\QG_Result.pnl Script: Initialize]:31

**Comment by mPokornyETM** [comment]
I found the same issue and can confirm, that there is some wrong logic. I am not sure now, where it was. But I know I fixed in my locale environment. Let me grab it and I will provide offical fix.

**Comment by mPokornyETM** [comment]
@dhoegerlETM can you pls assign this item to me? I have no permission for that. Otherwise I will forgot it in few minutes (you know - butterfly effect ;-) )

**Comment by mPokornyETM** [comment]
To be more precise. @dhoegerlETM  what you describe is that you can not see the result, but the issue mean the quality-gate will not check panels in sub directories (recursive ...)

**Comment by dhoegerlETM** [comment]
@mPokornyETM indeed I wanted to verify the finding but already failed in the first step: even without sub-directories I get the above mentioned log-errors and cannot see the results.

**Comment by mPokornyETM** [comment]
@dhoegerlETM  the mentioned failure will be fixed in #89 . That means you can see reports. but the recursive check is still not fixed.
